core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3307,11 +3298,8 @@ object Stream extends StreamLowPriority {
 
   /** Creates a random stream of integers using a random seed.
     */
-  def random[F[_]](implicit F: Sync[F]): Stream[F, Int] =
-    Stream.eval(F.delay(new scala.util.Random())).flatMap { r =>
-      def go: Stream[F, Int] = Stream.emit(r.nextInt()) ++ go
-      go
-    }
+  def random[F[_]](implicit F: Random[F]): Stream[F, Int] =
+    Stream.repeatEval(F.nextInt)
```

Comment:
I wonder if this is a worthy combinator, rather than something best left to users of `fs2` to run with the combinators.